src/App.tsx

```diff
@@ -1,10 +1,126 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState('');
+  const [newTodo, setNewTodo] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [loader, setLoader] = useState<number | null>();
+
+  const handleErrorMessage = (message: string) => {
+    setErrorMessage(message);
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(response => {
+        setTodos(response);
+        setErrorMessage('');
+      })
+      .catch(() => {
+        handleErrorMessage('Unable to load todos');
+      });
+  }, []);
+
+  function getFilterTodos(todoForFilter: Todo[], forFilter: string): Todo[] {
+    if (forFilter === 'active') {
+      return todoForFilter.filter(todo => todo.completed === false);
+    }
+
+    if (forFilter === 'completed') {
+      return todoForFilter.filter(todo => todo.completed === true);
+    }
+
+    return todoForFilter;
+  }
+
+  const filteredTodos = getFilterTodos(todos, filter);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!newTodo.trim()) {
+      setErrorMessage('Title should not be empty');
+
+      return;
+    }
+
+    const newTodoItem = {
+      title: newTodo,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    addTodo(newTodoItem)
+      .then(addedTodo => {
+        setTodos(currentTodos => [...currentTodos, addedTodo]);
+        setNewTodo('');
+        setErrorMessage('');
+      })
+      .catch(() => handleErrorMessage('Unable to add a todo'));
+  };
+
+  const handleDelete = (id: number) => {
+    setLoader(id);
+    deleteTodo(id)
+      .then(() => {
+        setTodos(currentTodo => currentTodo.filter(todo => todo.id !== id));
+        setErrorMessage('');
+      })
+      .catch(() => {
+        handleErrorMessage('Unable to delete a todo');
+      })
+      .finally(() => {
+        setLoader(null);
+      });
+  };
+
+  function handleClearCompleted() {
```

Comment:
It is important to follow the same way of creating functions everywhere, fix it everywhere
```suggestion
  const handleClearCompleted = () => {
```